File: RxTools-library/src/main/java/com/vondear/rxtools/view/scaleimage/decoder/DecoderFactory.java
package com.vondear.rxtools.view.scaleimage.decoder;

/**
 * @author vondear
 * @param <T> the class of decoder that will be produced.
 *
 * Interface for decoder (and region decoder) factories.
 */
public interface DecoderFactory<T> {
    /**
     * Produce a new instance of a decoder with type {@link T}.
     *
     * @return a new instance of your decoder.
     */
    T make() throws IllegalAccessException, InstantiationException;
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/scaleimage/decoder/DecoderFactory.kt
package com.tamsiree.rxui.view.scaleimage.decoder

/**
 * @author tamsiree
 * @param <T> the class of decoder that will be produced.
 *
 * Interface for decoder (and region decoder) factories.
</T> */
interface DecoderFactory<T> {
    /**
     * Produce a new instance of a decoder with type [T].
     *
     * @return a new instance of your decoder.
     */
    @Throws(IllegalAccessException::class, InstantiationException::class)
    fun make(): T
}